packages/api/src/index.ts

```diff
@@ -38,7 +38,13 @@ void (async () => {
           await entityManager.persistAndFlush(newUser);
         }
 
-        done(null, { profile, accessToken, refreshToken });
+        const guildIds = profile.guilds?.filter((guild) => guild.id === env.discordGuildId) ?? [];
+
+        if (guildIds.length > 0) {
+          done(null, { profile, accessToken, refreshToken });
+        } else {
+          done(new Error('Please make sure your are about of the event discord server'));
```

Comment:
Does this give a 403 automatically?

```suggestion
          done(new Error('Please make sure your are a member of the Discord server for this event'));
```

Reply:
It does not, I tried using the Discord ```AuthorizationError``` where it allowed me to attach the status/code but that threw an internal error of an uncaught exception for the error type.

Reply:
Hm, what if you create a new `PassportVerifyError` class that extends `Error`. Then you can pass that to `done`. You would need to set a new error handler though right before `app.listen` is called.

```typescript
export class PassportVerifyError extends Error {
  constructor(message: string) {
    super(message);
  }
}
```

```typescript
// After all other `app.use/get/post/etc` calls, before `app.listen`
app.use((req, res, next, err) => {
  if (err instanceof PassportVerifyError) {
    res.status(403).send(err.message);
    return;
  }

  next();
});
```